assets/section-main-product.css

```diff
@@ -1211,3 +1236,65 @@ a.product__text {
   margin-left: 1.2rem;
   flex-shrink: 0;
 }
+
+/* collapsed grid overrides */
+@media screen and (min-width: 750px) {
+  .product--stacked .grid--collapsed-y .grid__item:first-child {
+    --top-left-radius: var(--border-radius);
+    --top-right-radius: var(--border-radius);
+  }
+
+  .product--thumbnail .grid--collapsed .is-active,
+  .product--thumbnail_slider .grid--collapsed .is-active {
+    --top-left-radius: var(--border-radius);
+    --top-right-radius: var(--border-radius);
+    --bottom-right-radius: var(--border-radius);
+    --bottom-left-radius: var(--border-radius);
+  }
+}
+
+@media screen and (min-width: 750px) and (max-width: 989px) {
+  .product--stacked .grid--collapsed-y .grid__item:last-child,
+  .product--stacked .grid--collapsed-x:not(.grid--collapsed-y) .grid__item {
+    --bottom-right-radius: var(--border-radius);
+    --bottom-left-radius: var(--border-radius);
+  }
+
+  .product--stacked .grid--collapsed-x:not(.grid--collapsed-y) .grid__item {
+    --top-left-radius: var(--border-radius);
+    --top-right-radius: var(--border-radius);
+  }
+}
+
+@media screen and (min-width: 990px) {
+  .product--stacked .grid--collapsed-y .grid__item:first-child {
+    --top-left-radius: var(--border-radius);
+    --top-right-radius: var(--border-radius);
+  }
+
+  .product--stacked .grid--collapsed-y .grid__item:only-child,
+  .product--stacked .grid--collapsed-y .grid__item:last-child:nth-child(even) {
+    --bottom-left-radius: var(--border-radius);
+    --bottom-right-radius: var(--border-radius);
+  }
+
+  .product--stacked .grid--collapsed-y .grid__item:last-child:nth-child(odd) {
+    --bottom-right-radius: var(--border-radius);
+  }
+
+  .product--stacked .grid--collapsed-y .grid__item:nth-last-child(2):nth-child(even):not(.product__media-item--full) {
+    --bottom-left-radius: var(--border-radius);
+  }
+
+  .product--stacked .grid--collapsed-x:not(.grid--collapsed-y) .grid__item:first-child,
+  .product--stacked .grid--collapsed-x:not(.grid--collapsed-y) .grid__item:nth-child(even) {
+    --top-left-radius: var(--border-radius);
+    --bottom-left-radius: var(--border-radius);
+  }
+
+  .product--stacked .grid--collapsed-x:not(.grid--collapsed-y) .grid__item:nth-child(odd),
+  .product--stacked .grid--collapsed-x:not(.grid--collapsed-y) .grid__item:last-child:nth-child(even) {
+    --top-right-radius: var(--border-radius);
+    --bottom-right-radius: var(--border-radius);
+  }
+}
```

Comment:
I noticed some issues in here when the horizontal spacing is set to 0 and the vertical one is set to something greater than 0: 
![](https://screenshot.click/07-57-unij7-l0n9x.png)

And the other way around as well with some horizontal spacing and no vertical one: 
![](https://screenshot.click/07-10-dsfi8-v045l.png)